lldb-eval/value.cc

```diff
@@ -420,14 +484,20 @@ Value CastIntegerOrEnumToEnumType(lldb::SBTarget target, Value val, Type type) {
   return CreateValueFromAPInt(target, ext, type);
 }
 
-Value CastFloatToEnumType(lldb::SBTarget target, Value val, Type type) {
+Value CastFloatToEnumType(lldb::SBTarget target, Value val, Type type,
+                          Error& error) {
   assert(type.IsEnum() && "target type must be an enum");
   assert(val.type().IsFloat() && "argument must be a float");
 
   llvm::APSInt integer(type.GetByteSize() * CHAR_BIT, !type.IsSigned());
   bool is_exact;
   val.GetFloat().convertToInteger(integer, llvm::APFloat::rmTowardZero,
                                   &is_exact);
+
+  if (IsCastOutOfBounds(val, type)) {
```

Comment:
`APFloat::convertToInteger` returns `opStatus` (and has `is_exact` out parameter) -- https://github.com/llvm/llvm-project/blob/c162f086ba632ffaedfe92d63bf21571bc8ae4da/llvm/include/llvm/ADT/APFloat.h#L1113

Could we use it to detect invalid casts?

Reply:
The `is_exact` wasn't a good choice for floats with non-zero values after decimal point. I haven't checked the `opStatus` before, but it turns out that checking for the `opInvalidOp` flag seems to be enough in this case :)